mysk-data-api/src/routes/v1/subjects/electives/trade_offers/create_offer.rs

```diff
@@ -0,0 +1,256 @@
+                err,
+                "/subjects/electives/trade-offers".to_string(),
+            ));
+        }
+        Err(Error::InternalSeverError(err, _)) => {
+            return Err(Error::InternalSeverError(
+                err,
+                "/subjects/electives/trade-offers".to_string(),
+            ));
+        }
+        _ => unreachable!(),
+    };
+
+    // Checks if the sender is eligible to enroll in the receiver's elective session, also checks
+    // whether they're in a classroom
+    match DbElectiveSubject::is_student_eligible(
+        pool,
+        receiver_elective_subject.session_code,
+        sender_student_id,
+    )
+    .await
+    {
+        Ok(true) => (),
+        Ok(false) => {
+            return Err(Error::InvalidPermission(
+                "Student is not eligible to enroll in this elective".to_string(),
+                "/subjects/electives/trade-offers".to_string(),
+            ));
+        }
+        Err(Error::InvalidPermission(err, _)) => {
+            return Err(Error::InvalidPermission(
+                err,
+                "/subjects/electives/trade-offers".to_string(),
+            ));
+        }
+        _ => unreachable!(),
+    };
+
+    // Check if the sending student has an elective subject
+    let Some(sender_elective_subject_id) = query!(
+        r"
+        SELECT elective_subject_id FROM student_elective_subjects
+        WHERE student_id = $1 and year = $2 AND semester = $3
+        ",
+        sender_student_id,
+        get_current_academic_year(None),
+        get_current_semester(None),
+    )
+    .fetch_optional(pool)
+    .await?
+    else {
+        return Err(Error::InvalidPermission(
+            "Student does not have an elective subject".to_string(),
+            "/subjects/electives/trade-offers".to_string(),
+        ));
+    };
+    let sender_elective_subject_id = sender_elective_subject_id.elective_subject_id;
+
+    // Gets the elective subject of the sender
+    let sender_elective_subject = match ElectiveSubject::get_by_id_with_student_context(
+        pool,
+        sender_elective_subject_id,
+        sender_student_id,
+        Some(&FetchLevel::Compact),
+        None,
+    )
+    .await
+    {
+        Ok(ElectiveSubject::Compact(sender_elective_subject, _)) => sender_elective_subject,
+        Err(Error::EntityNotFound(err, _)) => {
+            return Err(Error::EntityNotFound(
+                err,
+                "/subjects/electives/trade-offers".to_string(),
+            ));
+        }
+        Err(Error::InternalSeverError(err, _)) => {
+            return Err(Error::InternalSeverError(
+                err,
+                "/subjects/electives/trade-offers".to_string(),
+            ));
+        }
+        _ => unreachable!(),
+    };
+
+    // Checks if the receiver is eligible to enroll in the sender's elective session
+    match DbElectiveSubject::is_student_eligible(
+        pool,
+        receiver_elective_subject.session_code,
```

Comment:
shouldn't this be `sender_elective_subject.session_code` so it check receiver against sender's elective

Reply:
Oops, will fix